data_collection/gazette/spiders/base.py

```diff
@@ -188,3 +188,63 @@ def get_gazette(self, document):
             file_urls=(document[0],),
             power="executive",
         )
+
+
+class ImprensaOficialSpider(BaseGazetteSpider):
+    @staticmethod
+    def _get_next_month(current_date):
+        month = (current_date.month % 12) + 1
+        year = current_date.year + (current_date.month + 1 > 12)
+        return date(year, month, 1)
+
+    @staticmethod
+    def _extract_public_url(file_url):
+        if not file_url:
+            return
+        # original: http://www.imprensaoficial.org/pdf/baixar.php?
+        #           arquivo=../pub/prefeituras/ba/ameliarodrigues/2020/proprio/1582.pdf
+        # download: http://www.imprensaoficial.org/pub/
+        #           prefeituras/ba/ameliarodrigues/2020/proprio/1582.pdf
+        index = file_url.find("arquivo=../") + 11
+        return f"http://www.imprensaoficial.org/{file_url[index:]}"
+
+    def start_requests(self):
+        current_date = self.start_date
+        while current_date <= date.today():
```

Comment:
**suggestion (non-blocking)**: Every loop iteration will call `date.today()` even if we are not expecting that this value changes between each iteration. It is not a big performance issue here, but if we have a costly function, we may start having problems.

```
end_date = date.today()
while current_date <= end_date:
    ...
```